tool/generate_samples_list.dart

```diff
@@ -0,0 +1,74 @@
+// ignore_for_file: avoid_print
+
+import 'dart:convert';
+import 'dart:io';
+
+/// Run from arcgis-maps-sdk-flutter-samples root directory
+/// dart run tool/generate_samples_list.dart
+void main() {
+  generateSamplesList();
+}
+
+/// Compiles all of the individual sample README.metadata.json files into a single JSON asset to feed into the Sample Viewer application.
+/// The output is in assets/generated_samples_list.json
+void generateSamplesList() {
+  final pathSeparator = Platform.pathSeparator;
+  final samples = <String, dynamic>{};
+
+  print('Generating samples list....');
+
+  // check for the required directories
+  final currentDirectory = Directory.current;
+  final samplesDirectory = Directory(
+      '${currentDirectory.path}${pathSeparator}lib${pathSeparator}samples');
```

Comment:
I wish there was a cleaner way to assemble platform-aware paths, but there just isn't. And actually here this might be overkill, because `Directory` is smart enough to understand `/` on both platforms. Unfortunately there's a minefield -- `Directory` and `File` are smart enough, but `Link` and `split()` aren't smart enough. Being explicit about the `pathSeparator` is definitely safest.

In another PR I used `Directory(Uri(pathSegments: ['..', '..', thing1, thing2]).toFilePath())` but I dunno if that's any better. Another option is something like `Directory([currentDirectory.path, 'libs', 'samples'].join(pathSeparator))`. I can't decide if we should standardize on which approach.

Reply:
Let's add it to our coding standards issue (I'll do that), and that way others can weigh in on any preferences.